src/plans/gradient_plan.jl

```diff
@@ -350,7 +401,36 @@ function (mg::MomentumGradient)(
 end
 
 """
-    AverageGradient <: DirectionUpdateRule
+    MomentumGradient()
+
+Append a momentum to a gradient processor, where the last direction and last iterate are
+stored and the new is composed as ``η_i = m*η_{i-1}' - s d_i``,
+where ``sd_i`` is the current (inner) direction and ``η_{i-1}'`` is the vector transported
+last direction multiplied by momentum ``m``.
+
+# Input
+
+* `M` (optional)
+
+# Keyword arguments
+
+* $(_kw_p_default)
+* `direction=`[`IdentityUpdateRule`](@ref) preprocess the actual gradient before adding momentum
+* $(_kw_X_default)`
+* `momentum=0.2` amount of momentum to use
+* $(_kw_vector_transport_method_default): $(_kw_vector_transport_method)
+
+!!! info
+    This function generates a [`DirectionUpdateRuleFactory`](@ref) for this
+    for the [`MomentumGradientRule`](@ref). If you do not provide the manifold
+    or any other of the keywords that require the manifold for a default,
+    you can provide a manifold later to use that manifold's default's instead.
```

Comment:
Here the message says "...for this for the..." which is a bit broken English. It also repeats in all other docstings, maybe worth to make a variable for this note and interpolate it in the same way as e.g. `$(_kw_p_default)`